src/emu/diimage.cpp

```diff
@@ -1309,10 +1309,9 @@ void device_image_interface::update_names()
 			count++;
 	}
 
-	m_canonical_instance_name = string_format("%s%d", inst_name, index + 1);
 	if (count > 1)
 	{
-		m_instance_name = m_canonical_instance_name;
```

Comment:
This whole function doesn't work right in edge cases.  You're only checking the long instance name for conflicts, not the brief instance name.  So you can end up with a situation where, for example, one device wants `promchip`/`prom` and another wants `promimage`/`prom` and you won't see it as a conflict so two devices end getting assigned `-prom` as their short option name.  You need to do it in a way that catches conflict groups on long or short name and renumbers accordingly.  It may require multiple passes.